### PR TITLE
[PLGN-321] - abnormal security- Add support to select the time filter field in get cases action

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - Added support to allow the user to select the time field that we filter on when using the get cases action as the old default was not supported
  - in the get threat details if `toAddress` comes back as a list as the customer has shown in the ticket to convert it to a comma separated string, to match the api docs and the behaviour we observe in our test instance 
  - to update the unit tests to reflect this new change 
  - bumped the sdk version to the latest 
  - bumped `dateparser` to 1.1.6
  - ran refresh to get the latest version of the schemas

https://issues.corp.rapid7.com/browse/PLGN-262
https://issues.corp.rapid7.com/browse/PLGN-321

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

new unit tests have been added for the `get_threat_details` actions as well and an update to `get_cases` action test to reflect the new code changes  

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/abnormal_security/Dockerfile [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
+ADD ./requirements.txt /python/src/requirements.txt
 
-# Install pip dependencies
 RUN if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
 
-# Install plugin
-RUN python setup.py build && python setup.py install
+ADD . /python/src
+
+RUN	python setup.py build && python setup.py install
```
Comment (joneill-r7) [inline plugins/abnormal_security/Dockerfile:15]: looks like strange spacing here?

File: plugins/abnormal_security/icon_abnormal_security/actions/get_cases/action.py [hunk 2]
```diff
@@ -13,6 +13,8 @@ def __init__(self):
     def run(self, params={}):
         return {
             Output.CASES: self.connection.api.get_cases(
-                from_date=params.get(Input.FROM_DATE, None), to_date=params.get(Input.TO_DATE, None)
+                from_date=params.get(Input.FROM_DATE, None),
+                to_date=params.get(Input.TO_DATE, None),
+                filter_key=params.get(Input.FILTER_KEY, None),
```
Comment (joneill-r7) [inline plugins/abnormal_security/icon_abnormal_security/actions/get_cases/action.py:18]: should this have a default if we don't require it as an input? Before it seemed to default to `receivedTime`, or maybe we pull the default from the schema in which case we don't need `None`. Don't need 'None' for the other input params either.

**Review by joneill-r7 — COMMENTED** [review]
Couple of small comments, but with changing the schema files we implement the `validate(response, NameOfTheActionOutput.schema)` into the unit tests as well

File: plugins/abnormal_security/Dockerfile (hunk 1 above)
Comment (rbowden-r7) [reply plugins/abnormal_security/Dockerfile:15]: I can update this to use a space, but this was added from the tooling from the looks of it 

image

File: plugins/abnormal_security/icon_abnormal_security/actions/get_cases/action.py (hunk 2 above)
Comment (rbowden-r7) [reply plugins/abnormal_security/icon_abnormal_security/actions/get_cases/action.py:18]: this has been updated to use `lastModifiedTime` as the default here, `receivedTime` no longer works for this endpoint so we are now giving the user options to select the field themselves

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**